File: services/src/test/java/com/senacor/code/fullstack/chat/rest/ChatMessageControllerTest.java
package com.senacor.code.fullstack.chat.rest;

import com.senacor.code.fullstack.chat.domain.ChatMessage;
import com.senacor.code.fullstack.chat.service.ChatMessageService;
import org.junit.Test;

import java.util.List;

import static java.util.Arrays.asList;
import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class ChatMessageControllerTest {

    private ChatMessageService serviceMock = mock(ChatMessageService.class);

    private ChatMessageController controller = new ChatMessageController(serviceMock);

    @Test
    public void loadChannelMessages() {
        List<ChatMessage> expected = asList(new ChatMessage(), new ChatMessage());
        when(serviceMock.loadChatMessages("dev")).thenReturn(expected);

        List<ChatMessage> result = controller.loadChatMessages("dev");

        assertEquals(expected, result);
        verify(serviceMock).loadChatMessages("dev");
    }

}

File: services/src/test/kotlin/com/senacor/code/fullstack/chat/message/ChatMessageControllerTest.kt
package com.senacor.code.fullstack.chat.message

import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import org.springframework.http.HttpStatus
import java.net.URI
import java.net.URISyntaxException

class ChatMessageControllerTest {

    private var serviceMock = mockk<ChatMessageService>()

    private var controller = ChatMessageController(serviceMock)

    @Test
    fun loadChannelMessages() {
        val expected = listOf(
            ChatMessage("dev", "s1", "m1"),
            ChatMessage("dev", "s2", "m2")
        )
        every { serviceMock.loadChatMessages("dev") } returns expected

        val result = controller.loadChatMessages("dev")

        assertEquals(expected, result)
        verify { serviceMock.loadChatMessages("dev") }
    }

    @Test
    @Throws(URISyntaxException::class)
    fun newChatMessagesTest() {
        val savedMessage = ChatMessage("dev", "sender@test.de", "Hello World!", id = "id-of-msg")
        every { serviceMock.saveChatMessage(any(), any(), any()) } returns savedMessage

        val result = controller.newChatMessages("dev", ChatMessage("dev", "sender@test.de", "Hello World!"))

        assertEquals(HttpStatus.CREATED, result. statusCode)
        assertEquals(URI("/api/channels/dev/messages/id-of-msg"), result.headers.location)
        verify { serviceMock.saveChatMessage("dev", "sender@test.de", "Hello World!") }
    }
}